wagtail/admin/tests/test_reports_views.py

```diff
@@ -412,6 +412,16 @@ def test_xlsx_export(self):
 
         self.assertEqual(worksheet["C2"].number_format, ExcelDateFormatter().get())
 
+    def test_report_renders_when_page_publisher_deleted(self):
+        temp_user = self.create_superuser(
+            "temp", email="temp@user.com", password="tempuser"
+        )
+        temp_user_id = temp_user.pk.db_value
```

Comment:
```suggestion
        temp_user_id = temp_user.pk
```

Reply:
This is kind of interesting and doesn't behave exactly as one might expect. The [primary key field](https://github.com/wagtail/wagtail/blob/b84825487381b3d5ac7bb0859924c20157227d61/wagtail/test/customuser/models.py#L56) for the custom user model is a [ConvertedValueField](https://github.com/wagtail/wagtail/blob/b84825487381b3d5ac7bb0859924c20157227d61/wagtail/test/customuser/fields.py#L38). When the `last_published_by` annotation is added to the queryset, it gets the `db_value` of the user's primary key, as it's not going through the `to_python` method of `ConvertedValueField`, so that is what ends up being rendered in the response. Anyway, accessing `db_value` on the `pk` is failing in at least one of the test environments, so I'll try a different approach.

Reply:
Ah, I see the reason for the failure - the failing case is using `emailuser.EmailUser` as the user class, which has a uuid field as its primary key.

Reply:
I wonder why django isn't able to use the correct field type here, as the subquery just uses the "user" foreign key, but maybe we could improve the `last_published_by`-annotation (e.g. by using "user_id" instead of user? or by using `output_field` on the Subquery? [docs](https://docs.djangoproject.com/en/3.2/ref/models/expressions/#django.db.models.Subquery)), because currently it would render the unmodified `db_value` on the frontend, which is also probably not what the developer expects.
So maybe we can use `temp_user_id = temp_user.pk` in the test at the end.

Reply:
When using `output_field` with the field type of the user model's primary key, it seems to have trouble prefetching the workflow states:

```
Traceback (most recent call last):
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/tests/test_reports_views.py", line 390, in test_xlsx_export
    response = self.get(params={"export": "xlsx"})
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/tests/test_reports_views.py", line 313, in get
    return self.client.get(reverse("wagtailadmin_reports:aging_pages"), params)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/test/client.py", line 836, in get
    response = super().get(path, data=data, secure=secure, **extra)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/test/client.py", line 424, in get
    return self.generic(
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/test/client.py", line 541, in generic
    return self.request(**r)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/test/client.py", line 810, in request
    self.check_exception(response)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/test/client.py", line 663, in check_exception
    raise exc_value
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/views/decorators/cache.py", line 62, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/urls/__init__.py", line 170, in wrapper
    return view_func(request, *args, **kwargs)
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/auth.py", line 205, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/views/generic/permissions.py", line 36, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/views/reports/base.py", line 26, in get
    self.object_list = self.decorate_paginated_queryset(self.object_list)
  File "/home/jmunn/projects/wagtail-dev/wagtail/wagtail/admin/views/reports/aging_pages.py", line 87, in decorate_paginated_queryset
    user_ids = set(queryset.values_list("last_published_by", flat=True))
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/db/models/query.py", line 320, in __iter__
    self._fetch_all()
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/db/models/query.py", line 1509, in _fetch_all
    self._prefetch_related_objects()
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/db/models/query.py", line 915, in _prefetch_related_objects
    prefetch_related_objects(self._result_cache, *self._prefetch_related_lookups)
  File "/home/jmunn/.virtualenvs/wagtail-dev/lib/python3.8/site-packages/django/db/models/query.py", line 1935, in prefetch_related_objects
    raise AttributeError(
AttributeError: Cannot find 'workflow_states' on ConvertedValue object, 'workflow_states' is an invalid parameter to prefetch_related()
```

The same problem does not occur when using the `EmailUser` model, where the primary key is a `UUIDField`. Using `user_id` instead of `user` for the annotation source still yields an int for CustomUser and str for EmailUser. Will need to dig a little deeper into why `ConvertedValue` is an issue for doing the prefetch, but for now I've added an explicit `to_python` call to the page processing.